MathLib/LinAlg/UnifiedMatrixSetters.cpp

```diff
@@ -97,6 +103,12 @@ void setVector(PETScVector& v,
     v.set(idcs, vals);
 }
 
+void setVector(PETScVector& v, MatrixVectorTraits<PETScVector>::Index const index,
```

Comment:
Indeed, thie functions introduced won't be used in many places. Furthermore, for performance reasons there maybe should be setters that take vectors if indices and values and set them all at once (future work, I commented on that in #1130).